etc/kernel-launchers/kubernetes/scripts/launch_kubernetes.py

```diff
@@ -96,15 +114,11 @@ def launch_kubernetes_kernel(
     kernel_class_name,
 ):
     """Launches a containerized kernel as a kubernetes pod."""
-
-    if os.getenv("KUBERNETES_SERVICE_HOST"):
-        config.load_incluster_config()
-    else:
-        config.load_kube_config()
-
     # Capture keywords and their values.
     keywords = dict()
 
+    kubernetes_client = KubernetesClientFactory().get_kubernetes_client()
```

Comment:
In general, I think the expectation is that the factory is initialized once (statically/atomically), and then it is invoked repeatedly to create other objects. I would define the factory statically somewhere (preferably in a superclass, but seeing none in a global constant), and then invoke it like this:

```suggestion
    kubernetes_client = KUBE_CLIENT_FACTORY.get_kubernetes_client()
```